lib/gyp/generator/ninja/index.js

```diff
@@ -140,6 +143,8 @@ Ninja.prototype.expand = function expand(p, productDir) {
     p = p.replace(/\$!INTERMEDIATE_DIR/g, intDir);
   }
 
+  // First one is in order to outsmart silly GYP file writers
+  p = p.replace(/(\.\.[\\\/])+\$\|CONFIGURATION_NAME/g, productDir);
```

Comment:
Because other than `ninja` the other generated output goes in the root dir, and artifacts go in `.../node/Release/...` or `..../node/Debug/...` and so `../../` in the path voodoo is base of root source (`.../node/`)